bftengine/src/bftengine/KeyManager.h

```diff
@@ -19,8 +21,13 @@ class KeyManager {
   };
 
  private:
-  bftEngine::IBasicClient* cl_{nullptr};
   int repID_{};
+  uint32_t clusterSize_{};
+  std::set<NodeIdType> exchangedReplicas_;  // TODO to/from Storage
+  std::string generateCid();
+  // Raw pointer is ok, since this class does not manage this resource.
+  InternalBFTClient* client_{nullptr};
```

Comment:
shared_ptr?

Reply:
same as above, if it's not an owner I don't see a reason for ownership semantic

Reply:
The whole point of using shared_ptrs is to safely manage lifetimes when multiple users reference the pointer. If only the owner/creator had it, then there would be no need for shared pointers to exist. In that case you'd just always use a unique_ptr and reference.

Reply:
There's actually likely a good argument that could be made that we don't need shared pointers here or in most our uses because we can manually track the lifetime and ownership easy enough. My guess is the reason it was specifically used in this case was because this object in particular is used across multiple threads.

Reply:
Shared pointer is for multiple users in different threads, where no specific owner is defined.
here the `InternalBFTClient` is tied to the lifetime of the replicaImp. there is no scenario where the KeyManager will own the resource exclusively. 
I think once a resource is being owned and managed, there is no need to use shared_ptr

Reply:
In that case, this is fine. Thanks for the explanation!